prov/sm2/src/sm2_common.h

```diff
@@ -86,45 +100,75 @@ enum {
  * 	src_data - src of additional op data (inject offset / resp offset)
  * 	data - remote CQ data
  */
-struct sm2_msg_hdr {
-	uint64_t msg_id;
-	int64_t id;
-	uint32_t op;
-	uint16_t op_src;
-	uint16_t op_flags;
-
-	uint64_t size;
-	uint64_t src_data;
+struct sm2_protocol_hdr {
+	/*
+	 * This is volatile for a reason, many things touch this
+	 * and we do not want compiler optimization here
+	 */
+	volatile long int next; /* fifo linked list next ptr*/
+	int64_t id;		/* id of msg sender*/
+	uint32_t op;		/* fi operation */
+	uint16_t op_src;	/* sm2 operation */
+	uint32_t op_flags;	/* flags associated with op */
+	uint64_t size;		/* Holds total size of message */
 	uint64_t data;
+	uint64_t tag; /* used for tagged messages */
+};
+
+struct sm2_free_queue_entry {
+	struct sm2_protocol_hdr protocol_hdr;
+	uint8_t data[SM2_INJECT_SIZE];
+};
+
+/*
+ * Unique smr_op_hdr for smr message protocol:
+ * 	addr - local shm_id of peer sending msg (for shm lookup)
+ * 	op - type of op (ex. ofi_op_msg, defined in ofi_proto.h)
+ * 	op_src - msg src (ex. smr_src_inline, defined above)
+ * 	op_flags - operation flags (ex. SMR_REMOTE_CQ_DATA, defined above)
+ * 	src_data - src of additional op data (inject offset / resp offset)
+ * 	data - remote CQ data
+ */
+struct sm2_msg_hdr {
```

Comment:
What's the difference between the sm2_protocol_hdr and the sm2_msg_hdr? Do you even use this? Is this something that was moved only to be removed in the next patch? And why isn't the formatting matching the clang format specification?

Reply:
Formatting is fixed. 

sm2_msg_header was essentially renamed sm2_protocol_header.  At one point, I thought I was going to have multiple data members relating to the Nemesis algorithm, and I wanted to group them all together.  It seems like I only need one, the next pointer.  I was planning on having:

```
sm2_msg_header {
   struct sm2_nemesis_hdr
   struct sm2_protocol_hdr
   char data[4096]
};
```

I don't think naming between sm2_protocol_hdr and sm2_msg_hdr should block the PR.

Reply:
reverted back to msg_hdr